test/services/etherpad/index.test.js

```diff
@@ -0,0 +1,164 @@
+	let app;
+	let testHelpers;
+	let configBefore;
+
+	before((done) => {
+		configBefore = Configuration.toObject();
+		freeport((err, port) => {
+			if (err) {
+				logger.warning('freeport:', err);
+			}
+
+			const API_PATH_CONFIG = Configuration.get('ETHERPAD_API_PATH');
+			const mockUrl = `http://localhost:${port}${API_PATH_CONFIG}`;
+			Configuration.set('ETHERPAD_URI', mockUrl);
+			Configuration.set('ETHERPAD_API_KEY', 'someapikey');
+
+			// eslint-disable-next-line global-require
+			app = require('../../../src/app');
+			server = app.listen(0);
+			testHelpers = testObjects(app);
+
+			const mock = MockServer(mockUrl, Configuration.get('ETHERPAD_API_PATH'), done);
+			mockServer = mock.server;
+		});
+	});
+
+	after(async () => {
+		await mockServer.close();
+		await server.close();
+		Configuration.reset(configBefore);
+	});
+
+	it('should create a new etherpad author', async () => {
+		const {
+			requestParams: { authentication: { accessToken } },
+		} = await testHelpers.setupUser({ roles: ['teacher'] });
+
+		const data = {};
+		const { body } = await request({
+			server: app,
+			method: 'post',
+			endpoint: '/etherpad/authors',
+			data,
+			accessToken,
+		});
+
+		expect(!!body.data.authorID).to.equal(true);
+	});
+
+
+	it('should create a new etherpad group', async () => {
+		const {
+			requestParams: { authentication: { accessToken } },
+		} = await testHelpers.setupUser({
+			roles: ['teacher'],
+			permissions: ['COURSE_VIEW']
+		});
+
+		const jwt = decode(accessToken);
+		const course = await testHelpers.createTestCourse({ teacherIds: [jwt.userId] });
+
+		const data = { courseId: course.id };
+
+		const { body } = await request({
+			server: app,
+			method: 'post',
+			endpoint: '/etherpad/groups',
+			data,
+			accessToken,
+		});
+
+		expect(!!body.data.groupID).to.equal(true);
+	});
+
+	it('should create a new pad', async () => {
+		const {
+			requestParams: { authentication: { accessToken } },
+		} = await testHelpers.setupUser({ roles: ['teacher'] });
+
+		const jwt = decode(accessToken);
+		const course = await testHelpers.createTestCourse({ userIds: [jwt.userId] });
+
+		const data = {
+			courseId: course.id,
+			padName: 'test',
+			text: 'Grande Mucho',
+		};
+
+		const { body } = await request({
+			server: app,
+			method: 'post',
+			endpoint: '/etherpad/pads',
+			data,
+			accessToken,
+		});
+
+		expect(!!body.data.padID).to.equal(true);
+	});
+
+	it('should create a new session', async () => {
+		const {
+			requestParams: { authentication: { accessToken } },
+		} = await testHelpers.setupUser({ roles: ['teacher'] });
+
+		const jwt = decode(accessToken);
+		const course = await testHelpers.createTestCourse({ userIds: [jwt.userId] });
+
+		const data = {
+			courseId: course.id,
+		};
+
+		const { body } = await request({
+			server: app,
+			method: 'post',
+			endpoint: '/etherpad/sessions',
+			data,
+			accessToken,
+		});
+
+		expect(!!body.data.sessionID).to.equal(true);
+	});
+
+});
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Block must not be padded by blank lines.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5525245)